src/pybind/mgr/dashboard/module.py

```diff
@@ -565,7 +587,19 @@ def clients(self, fscid_str):
             @cherrypy.expose
             @cherrypy.tools.json_out()
             def clients_data(self, fs_id):
-                return self._clients(int(fs_id))
+                try:
+                    fs_id = int(fs_id)
+                except ValueError:
+                    raise cherrypy.HTTPError(400,
+                        "Invalid filesystem id {0}".format(fs_id))
+
+                try:
+                    r = self._clients(fs_id)
+                except TypeError:
```

Comment:
Should update _clients to handle the RemoteViewCache.VALUE_NONE status instead so that we don't have to handle TypeError here (where the comment "TODO do something sensible with status" is in `_clients`)